File: micrometer-core/src/test/groovy/io/micronaut/docs/OutOfStockException.java
package io.micronaut.docs;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.HttpResponseProvider;

public class OutOfStockException extends RuntimeException implements HttpResponseProvider {

    @Override
    public HttpResponse<?> getResponse() {
        return HttpResponse.ok(0);
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/metrics/exception/OutOfStockException.kt
package io.micronaut.docs.metrics.exception

// tag::imports[]
import io.micronaut.http.HttpResponse
import io.micronaut.http.HttpResponseProvider
// end::imports[]

// tag::class[]
class OutOfStockException : RuntimeException(), HttpResponseProvider {

    override fun getResponse(): HttpResponse<*> = HttpResponse.ok(0)
}
// end::class[]
